contents/javascript/Javascript-framework-Angular.md

```diff
@@ -0,0 +1,226 @@
+You will notice that as you change the name in the text box, the name rendered on the app is also updated. This is because the UI and the application data are automatically synchronized by Angular as explained [here](https://angular.io/guide/template-syntax#binding-syntax-an-overview).
+
+Angular organizes the behavior instructions (written in the TypeScript language) into what it calls _Components_. Given below is the component file `person.component.ts` of the `PersonComponent` :  
+
+```typescript
+@Component({
+  selector: 'app-person',
+  templateUrl: './person.component.html',
+  styleUrls: ['./person.component.css']
+})
+export class PersonComponent {
+  person: Person = {
+    name: 'John Doe'
+  };
+  constructor() { }
+}
+```
+
+The `@Component` tag is used to mark a component. It provides the metadata that determines how the component is processed during runtime.
+
+* The use of `selector: app-person` is how Angular identifies this component in a template. The following code constitutes the template of `app.component.html`.
+
+    ```html
+    <h1>Person Data</h1>
+    <app-person></app-person>
+    <hr/>
+    ```
+
+    Angular identifies the `<app-person>` tag, instantiates the `PersonComponent` and renders the view defined by the template `./person.component.html`.
+
+* The `person.name` used in the template `./person.component.html`, refers to the name property defined in the above component.
+
+Now let us look at what **Dependency Injection**(DI) means. Dependencies are services that a class needs in order to perform it's function. We can visualize this with the help of an example.
+
+<center>
+<img src="dependency.png" width="60%">
+
+_Figure 1. What is a dependency?_ <sup>[source](https://medium.freecodecamp.org/a-quick-intro-to-dependency-injection-what-it-is-and-when-to-use-it-7578c84fa88f)</sup>
+</center>
+
+Dependency injection is a way of providing a class with the required services. The official website claims that the DI framework increases the "efficiency and modularity" of an app. Given below is a an injectable service class.
+ 
+```typescript
+import { Injectable } from '@angular/core';
+import { PERSONS } from './mock-persons';
+@Injectable({
+  providedIn: 'root',
+})
+export class PersonService {
+  getPersons() { return PERSONS; }
+}
+```
+
+The `PersonComponent` declared previously, can make use of this `PersonService` by injecting the dependency in its constructor as a parameter.
+
+```
+constructor(personService: PersonService) {
+ this.persons = personService.getPersons();
+}
+```
+
+In the above example, we write the `getPersons` method only once in the service class and we can access this method from any component by injecting the service in the constructor.
+This explains how DI increases "efficiency and modularity".
```

Comment:
Actually why is efficient and modularity double quoted? Can consider adding a tooltip instead to explain these terms so you dont have to double quote it?

Also, is shows a better word than explains since you are mostly showcasing an example

Reply:
It is double-quoted because it is taken from the official website.